src/server/req_runjob.c

```diff
@@ -1048,15 +1012,12 @@ svr_strtjob2(job *pjob, struct batch_request *preq)
 	}
 
 	/* send the job to MOM */
-	if (!(pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob))
-	{
-		job_attr_def[(int) JOB_ATR_Comment].at_decode(
-				&pjob->ji_wattr[(int) JOB_ATR_Comment],
-				NULL,
-				NULL,
-				form_attr_comment("Job was sent for execution at %s",
-						pjob->ji_wattr[(int) JOB_ATR_exec_vnode].at_val.at_str));
-	}
+	job_attr_def[(int) JOB_ATR_Comment].at_decode(
```

Comment:
As discussed the right way of setting value of an attribute is to use the function set_attr_svr() which takes care of clearing,allocating memory for the attribute etc. Please use this function instead of calling decode() directly.

Reply:
fixed in rc 23 https://github.com/PBSPro/pbspro/pull/590/commits/75a2cfae3e21fc17658ef792e406c158ecdc6314